### PR TITLE
Packaging and installation improvements

### PR DESCRIPTION
**Pull request by clebergnu**
This collection of patches tries to streamline the installation experience that users will get with the standard Python tools.

In short, we rename the *package* to `avocado-framework`, adjust some optional packages from `requires*.txt` and finally add core requirements to `setup.py` itself.

This should allow any user to install avocado using `pip install avocado-framework` after this is incorporated and a new package is pushed to PyPI.  Until then, the same experience can be tested by running:

```
pip install -e git://github.com/clebergnu/avocado.git@requirements#egg=avocado-framework
```

This can be done, say, in a virtual environment, and the basic avocado functionality should be up and running with this single command.

### CODE REVIEW

**Comment by lmr** [comment]
Bravo, Cleber! This will make the install of avocado via pip so much easier.

I'll try to review this, but I'm in Scylla Dev Summit right now, so it might take a while.

File: setup.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def get_long_description():
     return req_contents
 
 if __name__ == '__main__':
-    setup(name='avocado',
+    setup(name='avocado-framework',
```
Comment (lmr) [inline setup.py]: OK

File: optional_plugins/html/setup.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
       url='http://avocado-framework.github.io/',
       packages=find_packages(),
       include_package_data=True,
-      install_requires=['pystache'],
+      install_requires=['avocado-framework', 'pystache'],
```
Comment (lmr) [inline optional_plugins/html/setup.py]: OK

File: requirements-docs.txt [hunk 3]
```diff
@@ -1,6 +1,5 @@
 # All pip installable requirements pinned for readthedocs.org
 fabric==1.10.0
-pystache==0.5.4
```
Comment (lmr) [inline requirements-docs.txt]: OK

File: setup.py [hunk 4]
```diff
@@ -149,4 +149,5 @@ def get_long_description():
               },
           zip_safe=False,
           test_suite='selftests',
-          python_requires='>=2.6')
+          python_requires='>=2.6',
+          install_requires=['stevedore'])
```
Comment (lmr) [inline setup.py]: The problem is, when someone installs from pip, if not all the deps are properly stated, we might end up with a lot of broken functionality...

Comment (lmr) [inline setup.py]: That, of course, assuming we are on a very bare bones system.

Comment (lmr) [inline setup.py]: To show what I mean, I just installed from your branch into a newly created virtual environment:

    $ pip install -e git://github.com/clebergnu/avocado.git@requirements#egg=avocado-framework
    Obtaining avocado-framework from git+git://github.com/clebergnu/avocado.git@requirements#egg=avocado-framework
      Cloning git://github.com/clebergnu/avocado.git (to requirements) to ./lmr-test-avocado-changes/src/avocado-framework
      Running setup.py egg_info for package avocado-framework
        /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'use_2to3'
          warnings.warn(msg)
        /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'python_requires'
          warnings.warn(msg)
        
    Downloading/unpacking stevedore (from avocado-framework)
      Downloading stevedore-1.17.1.tar.gz (501kB): 501kB downloaded
      Running setup.py egg_info for package stevedore
        
        Installed /home/lmr/lmr-test-avocado-changes/build/stevedore/pbr-1.10.0-py2.7.egg
        [pbr] Processing SOURCES.txt
        warning: LocalManifestMaker: standard file '-c' not found
        
        warning: no previously-included files found matching '.gitignore'
        warning: no previously-included files found matching '.gitreview'
        warning: no previously-included files matching '*.pyc' found anywhere in distribution
        warning: no files found matching '*.rst' under directory 'docs'
        warning: no files found matching '*.py' under directory 'docs'
        warning: no files found matching '*.html' under directory 'docs'
        warning: no files found matching '*.css' under directory 'docs'
        warning: no files found matching '*.js' under directory 'docs'
        warning: no files found matching '*.png' under directory 'docs'
        warning: no files found matching '*.txt' under directory 'docs'
        warning: no files found matching '*.jpg' under directory 'docs'
        warning: no files found matching '*.py' under directory 'tests'
    Downloading/unpacking pbr>=1.6 (from stevedore->avocado-framework)
      Downloading pbr-1.10.0.tar.gz (111kB): 111kB downloaded
      Running setup.py egg_info for package pbr
        [pbr] Processing SOURCES.txt
        warning: LocalManifestMaker: standard file '-c' not found
        
        warning: no previously-included files found matching '.gitignore'
        warning: no previously-included files found matching '.gitreview'
        warning: no previously-included files matching '*.pyc' found anywhere in distribution
    Downloading/unpacking six>=1.9.0 (from stevedore->avocado-framework)
      Downloading six-1.10.0.tar.gz
      Running setup.py egg_info for package six
        
        no previously-included directories found matching 'documentation/_build'
    Installing collected packages: avocado-framework, stevedore, pbr, six
      Running setup.py develop for avocado-framework
        /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'use_2to3'
          warnings.warn(msg)
        /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'python_requires'
          warnings.warn(msg)
        
        Creating /home/lmr/lmr-test-avocado-changes/lib/python2.7/site-packages/avocado-framework.egg-link (link to .)
        Adding avocado-framework 40.0 to easy-install.pth file
        Installing avocado script to /home/lmr/lmr-test-avocado-changes/bin
        Installing avocado-rest-client script to /home/lmr/lmr-test-avocado-changes/bin
        
        Installed /home/lmr/lmr-test-avocado-changes/src/avocado-framework
      Running setup.py install for stevedore
        [pbr] Generating AUTHORS
        [pbr] AUTHORS complete (0.0s)
        [pbr] Reusing existing SOURCES.txt
      Running setup.py install for pbr
        [pbr] Generating AUTHORS
        [pbr] AUTHORS complete (0.0s)
        [pbr] Reusing existing SOURCES.txt
        Installing pbr script to /home/lmr/lmr-test-avocado-changes/bin
      Running setup.py install for six
        
        no previously-included directories found matching 'documentation/_build'
    Successfully installed avocado-framework stevedore pbr six
    Cleaning up...
    $ avocado --version
    Avocado 40.0
    Failed to load plugin from module "avocado.plugins.remote": ImportError('No module named fabric.exceptions',)
    Failed to load plugin from module "avocado.plugins.vm": ImportError('No module named fabric.exceptions',)
    Failed to load plugin from module "avocado.plugins.docker": ImportError('No module named aexpect',)

Comment (lmr) [inline setup.py]: So yeah, we don't have crashes anymore, which is *awesome*, but still the experience could be a lot better.

Comment (clebergnu) [inline setup.py]: Actually, that was the intended scope of this first round.  We're working towards making remote, vm, docker and all of those optional plugins, with their own deps declared in their own `setup.py`.

And btw, thanks so much for testing it!

Comment (ldoktor) [inline setup.py]: Impressive, that's expected outcome. I'm looking forward having `remote` as separate plugin (the same way we have the `html` plugin.

File: requirements-docs.txt (hunk 3 above)
Comment (ldoktor) [inline requirements-docs.txt]: This reminded me, that we don't test `html` plugin automatically. I double checked by running coverage and yes, unless one uses `make link` it's not involved.

On the other hand we test it manually before release, but I think it ought to be added to travis check, even if only as `make link && make smokecheck` after each check.

Comment (ldoktor) [inline requirements-docs.txt]: Btw that issue is not related to this code, this patch is fine :+1:

File: setup.py (hunk 4 above)
Comment (ldoktor) [inline setup.py]: Actually after reading https://packaging.python.org/requirements/ I think we should list all the essential packages required for avocado, until we make them optional. Also we should define the version as eg. we do require certain version of stevedore (not sure if conditional versions are supported, but we can add it conditionally).

Comment (ldoktor) [inline setup.py]: PS: I don't mean to copy whole `requirements`, but only to put there all dependencies in order to allow basic functionality without errors (we don't need PyYAML, libvirt-python, pyliblzma...)

File: setup.py (hunk 1 above)
Comment (ldoktor) [inline setup.py]: It's a bit strange as one still imports it as `avocado`, which is a bit odd, but it seems to be working well.

Comment (clebergnu) [inline setup.py]: Yeah, that's one of the misconceptions about package names and the module names, which kept me from doing this before. They're actually very different things.

File: setup.py (hunk 4 above)
Comment (ldoktor) [inline setup.py]: OK after discussion with @clebergnu we agree that this is an improvement and we don't want it to cover everything (especially the python 2.6 with many custom dependencies).

As we don't handle 2.6 we should not specify 2.7 specific stevedore version so let's just keep it as is (without any specific version required) and leave it on python 2.6 users to use requirements.

The ImportErrors mentioned by @lmr  are expected as plugins are expected to warn about malfunctioning, but the core functionality is fine.
